README.md

```diff
@@ -1 +1,121 @@
 # xtractmime
+
+`xtractmime` is a [BSD-licensed](https://opensource.org/licenses/BSD-3-Clause)
+Python 3.6+ implementation of the [MIME Sniffing
+Standard](https://mimesniff.spec.whatwg.org/).
+
+Install from [`PyPI`](https://pypi.python.org/pypi/xtractmime):
+
+```
+pip install xtractmime
+```
+
+---
+
+## Basic usage
+
+Below mentioned are some simple examples of using `xtractmime.extract_mime`:
+
+```python
+>>> from xtractmime import extract_mime
+>>> extract_mime(b'Sample text content')
+b'text/plain'
+>>> extract_mime(b'', content_types=(b'text/html',))
+b'text/html'
+```
+
+Additional functionality to check if a MIME type belongs to a specific MIME type group using 
+methods mentioned in `xtractmime.mimegroups`:
+
+```python
+>>> from xtractmime.mimegroups import is_html_mime_type, is_image_mime_type
+>>> mime_type = b'text/html'
+>>> is_html_mime_type(mime_type)
+True
+>>> is_image_mime_type(mime_type)
+False
+```
+
+---
+
+## API Reference
+
+### function `xtractmime.extract_mime : Optional[bytes]`
+**Parameters:**
+
+* `body: bytes`
+* `content_types: Optional[Tuple[bytes]], default = None`
+* `http_origin: bool, default = True`
+* `no_sniff: bool, default = False`
+* `extra_types: Optional[Tuple[Tuple[bytes, bytes, Optional[Set[bytes]], bytes], ...]], default = None`
+* `supported_types: Set[bytes], default = None`
```

Comment:
No need for the `, default` part, e.g. `content_types: Optional[Tuple[bytes]] = None` should be enough.

Also, I wonder why the typing check does not see this, but the type for `supported_types` should be `Optional[Set[bytes]]`. This would need an update to the code as well.

Reply:
> Also, I wonder why the typing check does not see this, but the type for `supported_types` should be `Optional[Set[bytes]]`. This would need an update to the code as well.

I had a doubt on this earlier but the typing checks were passing so I didn't mention it. It is still passing though after changing it to `Optional[Set[bytes]]`